frontend/apps/app/features/organizations/pages/OrganizationDetailPage/OrganizationDetailPage.tsx

```diff
@@ -0,0 +1,194 @@
+'use client'
+
+import { createClient } from '@/libs/db/client'
+import { urlgen } from '@/utils/routes'
+import { Button } from '@liam-hq/ui'
+import { useRouter } from 'next/navigation'
+import { type FC, type FormEvent, useState } from 'react'
+import { Tab, Tabs } from '../../components/Tabs'
+import styles from './OrganizationDetailPage.module.css'
+
+interface OrganizationDetailPageProps {
+  organization: {
+    id: number
+    name: string
+  }
+  members: Array<{
+    id: number
+    user: {
+      id: string
+      name: string
+      email: string
+    }
+  }> | null
+  invites: Array<{
+    id: number
+    email: string
+    inviteBy: {
+      id: string
+      name: string
+      email: string
+    }
+  }> | null
+}
+
+export const OrganizationDetailPage: FC<OrganizationDetailPageProps> = ({
+  organization,
+  members,
+  invites,
+}) => {
+  const [name, setName] = useState(organization.name)
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState<string | null>(null)
+  const [successMessage, setSuccessMessage] = useState<string | null>(null)
+  const router = useRouter()
+
+  const handleUpdate = async (e: FormEvent) => {
+    e.preventDefault()
+    if (!name) {
+      setError('Organization name is required')
+      return
+    }
+
+    setLoading(true)
+    setError(null)
+    setSuccessMessage(null)
+
+    try {
+      const supabase = await createClient()
+
+      const { error: updateError } = await supabase
+        .from('Organization')
+        .update({ name })
+        .eq('id', organization.id)
```

Comment:
In final implementation, we do not want to access supabase on the client side, but only Server Actions.